### PR TITLE
[Doc] Add Schema Merge examples to Files

### PR DESCRIPTION
**Pull request by EsoragotoSpirit**
Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>

## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/en/sql-reference/sql-functions/table-functions/files.md [hunk 1]
```diff
@@ -163,6 +163,8 @@ After the sampling, StarRocks unionizes the columns from all the data files acco
 - For columns with the same column name but different data types, they are identified as the same column but with a general data type on a relative fine granularity level. For example, if the column `col1` in file A is INT but DECIMAL in file B, DOUBLE is used in the returned column.
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-functions/table-functions/files.md:163]: We should give the specific union rules.
1. All INTEGERs will be unioned as the larger INTEGER.
2.An INTEGER column and a FLOAT column will be unioned as a DECIMAL column.
3. Others to STRING.

File: docs/en/sql-reference/sql-functions/table-functions/files.md [hunk 2]
```diff
@@ -349,3 +351,168 @@ FILES(
...
+104,3,2018-05-15T18:30:00
+105,6,2018-05-15T18:30:00
+106,1,2018-05-15T18:30:00
+107,8,2018-05-15T18:30:00
+108,5,2018-05-15T18:30:00
+109,6,2018-05-15T18:30:00
+110,8,2018-05-15T18:30:00
+```
+
+Create a table named `test_ctas_parquet` and insert the data rows from the two Parquet files into the table:
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-functions/table-functions/files.md:391]: Use a CTAS statement to create xxx

File: docs/en/sql-reference/sql-functions/table-functions/files.md [hunk 3]
```diff
@@ -349,3 +351,168 @@ FILES(
...
+
+```SQL
+CREATE TABLE test_ctas_parquet AS
+SELECT * FROM FILES(
+        "path" = "s3://inserttest/parquet/*",
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-functions/table-functions/files.md:396]: 4 spaces indent

File: docs/en/sql-reference/sql-functions/table-functions/files.md [hunk 4]
```diff
@@ -349,3 +351,168 @@ FILES(
...
+Create Table: CREATE TABLE `test_ctas_parquet` (
+  `$1` bigint(20) NULL COMMENT "",
+  `$2` decimal(38, 9) NULL COMMENT "",
+  `$3` varchar(1048576) NULL COMMENT ""
+) ENGINE=OLAP 
+DUPLICATE KEY(`$1`, `$2`)
+COMMENT "OLAP"
+DISTRIBUTED BY RANDOM
+PROPERTIES (
+"bucket_size" = "4294967296",
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-functions/table-functions/files.md:422]: 4 spaces indent

**Review by wangsimo0 — APPROVED** [review]

**Review by jaogoy — APPROVED** [review]

**Pull request merged**